File: 2st/Riyeon/배열 만들기 6.java
import java.util.*;
class Solution {
    public int[] solution(int[] arr) {
        List<Integer> stk = new ArrayList<>();
        for(int i=0; i<arr.length; i++){
            if(stk.isEmpty())
                stk.add(arr[i]);
            else if(stk.get(stk.size()-1) == arr[i])
                stk.remove(stk.size()-1);
            else
                stk.add(arr[i]);
        }
        
        if(stk.isEmpty())
            return new int[] {-1};
        else
            return stk.stream().mapToInt(i->i).toArray();
    }
}

File: 2st/Yeonwoo/배열 만들기 6.py
def solution(arr):
    answer = []
    i=0
    while i < len(arr):
        if answer == []:
            answer.append(arr[i])
            i+=1
        elif answer[-1] == arr[i]:
            answer.pop()
            i+=1
        elif answer[-1]!=arr[i]:
            answer.append(arr[i])
            i+=1
    if answer==[]:
        answer.append(-1)
    
    return answer